### PR TITLE
[callbacks] Fix memory leak in callbacks (debug build only)

### PR DESCRIPTION
**Pull request by jprestwo**
Signed-off-by: James Prestwood <james.prestwood@intel.com>

### CODE REVIEW

**Comment by jprestwo** [comment]
@grgustaf, per our discussion, don't merge this just yet.

File: src/zjs_callbacks.c [hunk 1]
```diff
@@ -95,16 +96,23 @@ static int zjs_ringbuf_error_max = 0;
 static int zjs_ringbuf_last_error = 0;
 
 #ifdef DEBUG_BUILD
-static void set_info_string(char **str, const char *file, const char *func)
+static void set_info_string(char *str, const char *file, const char *func)
 {
-    uint32_t file_len = strlen(file);
+    char *i = (char*)(file + strlen(file));
```
Comment (grgustaf) [inline src/zjs_callbacks.c]: Just for consistency do (char *)

Comment (grgustaf) [inline src/zjs_callbacks.c]: Wait, actually this should be const char *i on the left and you shouldn't need a cast on the right.

Comment (grgustaf) [inline src/zjs_callbacks.c]: I confused myself earlier but yes as I said I think you should be using const char *i on the left, and not casting on the right.

Better yet come up with a variable name :D.

File: src/zjs_callbacks.c [hunk 2]
```diff
@@ -95,16 +96,23 @@ static int zjs_ringbuf_error_max = 0;
...
-    *str = zjs_malloc(file_len + func_len + 4);
-    if (!(*str)) {
-        ERR_PRINT("failed to create file/func info string\n");
-        return;
+    uint32_t total = file_len + func_len + 4;
+    if (total > MAX_CALLER_CREATOR_LEN) {
+        total = MAX_CALLER_CREATOR_LEN;
     }
-    snprintf(*str, file_len + func_len + 4, "%s:%s()", file, func);
+    snprintf(str, file_len + func_len + 4, "%s:%s()", i, func);
```
Comment (grgustaf) [inline src/zjs_callbacks.c]: You meant to use `total` here; however you can just pass MAX_CALLER_CREATOR_LEN all the time instead, and not bother calculating `total`.

**Comment by grgustaf** [comment]
+1

**Pull request merged**